uiowa/uiowa01
# Feature: Replace Lightning Media with custom-built components

Finally...

This feature allows us to be able to remove Lightning components by replacing the elements that we were depending on for other custom modules or the collegiate_basic profile. For better or worse, this is a fairly big change and testing it is a bit tricky.

Some things to note:
* This update doesn't actually remove the Lightning components, yet. That will come after in a separate feature.
* Instead, this creates media components to replace the ones provided by Lightning and swaps those out in the collegiate_basic profile.
* To make it easier to do a side-by-side comparison, I created a clone of collegiate_basic - called collegiate_basic_old - which can be used to do a site install for another site in case you want to see what the settings or UI were like before the change.
* After we have tested this PR, I will remove the collegiate_basic_old profile and push a new commit.

# Tests

## 1. Installation
`drush @theming.local si collegiate_basic` to ensure no issues with installation.

## 2. Check the media types
Confirm that all media types are present after installation: Document, Image, Video, Remote Video.

## 3. Test block types
* Create a basic page piece of content and edit in layout builder. 
* Add all the block types provided by uiowa_lb_blocks.
* Confirm that media fields work as expected.

## 4. Test media fields on entity_types_* content types
* Install each entity_types_* module.
* Create a piece of content for each content type.
* Confirm that media fields work as expected.

## 5. Test a multi-media field
* Create a new media field on the basic page content type.
* _Note:_ When creating the media field, you may get a big red error message. Feel free to document - this is a known issue that I'm hoping is fixed by updating the Drupal core version later.
* Select at least the Image, Video, and Remote Video media types and save the field settings.
* Create a piece of basic page content and use the new field.
* Confirm that each media type widget works as expected.
